eng/scripts/Pack-Crates.ps1

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env pwsh
+
+#Requires -Version 7.0
+param(
+  [string]$OutputPath,
+  [Parameter(ParameterSetName = 'Named')]
+  [string[]]$PackageNames,
+  [Parameter(ParameterSetName = 'PackageInfo')]
+  [string]$PackageInfoDirectory
+)
+
+$ErrorActionPreference = 'Stop'
+
+. (Join-Path $PSScriptRoot '..' 'common' 'scripts' 'common.ps1')
+. (Join-Path $EngCommonScriptsDir 'Helpers' 'CommandInvocation-Helpers.ps1')
+
+if ($OutputPath) {
+  $OutputPath = New-Item -ItemType Directory -Path $OutputPath -Force | Select-Object -ExpandProperty FullName
+}
+
+function Get-OutputPackageNames($workspacePackageNames) {
```

Comment:
Just based on the name, shouldn't this filter out `publish = false` as well?

Reply:
Will create an issue for future PR